app-games/src/main/java/com/aptoide/android/aptoidegames/permissions/AppPermissionsManager.kt

```diff
@@ -0,0 +1,67 @@
+package com.aptoide.android.aptoidegames.permissions
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.provider.Settings
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import cm.aptoide.pt.extensions.hasNotificationsPermission
+import com.aptoide.android.aptoidegames.permissions.AppPermissionsManager.PreferencesKeys.NOTIFICATION_PERMISSION_REQUESTED
+import com.aptoide.android.aptoidegames.permissions.AppPermissionsManager.PreferencesKeys.STORAGE_PERMISSION_REQUESTED
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+class AppPermissionsManager @Inject constructor(
```

Comment:
Looks like a candidate to spilt into a single interface and 2 implementations...